dasgo/resources.go

```diff
@@ -135,20 +137,27 @@ func (c TextInput) Type() Flag {
 	return FlagComponentTypeTextInput
 }
 
+// TODO: was unclear on optionality and nullablility so I dropped omitempty.
+
 // https://discord.com/developers/docs/interactions/message-components#component-object
 type ActionsRow struct {
-	Components []Component `json:"components,omitempty"`
+	Components []Component `json:"components"`
 }
 
+
+
 // Button Object
+
+// *The type field was not included
```

Comment:
Comment needs clarification.

Reply:
The table from the url had a field called type that was not in the go file. Should I add the type field?

Reply:
<h3 class="h3-CT8zp6" id="button-object" style="font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; margin: 25px 0px 4px; padding: 0px; border: 0px; outline: 0px; font-weight: 600; font-family: Whitney, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 20px; vertical-align: baseline; display: flex; color: rgb(255, 255, 255); line-height: 24px;">Button Object<a class="anchor-1MIwyf hyperlink" focusprops="[object Object]" href="https://discord.com/developers/docs/interactions/message-components#button-object" role="button" tabindex="0" style="margin: 0px 0px 0px 10px; padding: 0px; border: 0px; outline: 0px; font-weight: inherit; font-style: inherit; font-family: inherit; font-size: 22px; vertical-align: baseline; text-decoration: none; color: hsl(197,calc(var(--saturation-factor, 1)*100%),47.8%); cursor: pointer; transition: color 0.1s; border-top-left-radius: 4px; border-top-right-radius: 4px; border-bottom-right-radius: 4px; border-bottom-left-radius: 4px; opacity: 0; align-self: center; background-image: url(&quot;/assets/40e8a44f9e49d1d30b81b03f7065b738.svg&quot;); width: 14px; height: 14px;"><div name="button-object" style="margin: 0px; padding: 0px; border: 0px; outline: 0px; font-weight: inherit; font-style: inherit; font-family: inherit; font-size: 22px; vertical-align: baseline;"></div></a></h3><h6 class="h6-2Fz8hG" id="button-object-button-structure" style="font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; margin: 24px 0px 8px; padding: 0px; border: 0px; outline: 0px; font-weight: 600; font-family: Whitney, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 16px; vertical-align: baseline; display: flex; -webkit-box-align: center; align-items: center; background-color: rgb(54, 57, 63); border-top-left-radius: 3px; border-top-right-radius: 3px; color: rgb(227, 229, 232); height: 36px;">Button Structure<a class="anchor-1MIwyf hyperlink" focusprops="[object Object]" href="https://discord.com/developers/docs/interactions/message-components#button-object-button-structure" role="button" tabindex="0" style="margin: 0px 0px 0px 10px; padding: 0px; border: 0px; outline: 0px; font-weight: inherit; font-style: inherit; font-family: inherit; font-size: 22px; vertical-align: baseline; text-decoration: none; color: hsl(197,calc(var(--saturation-factor, 1)*100%),47.8%); cursor: pointer; transition: color 0.1s; border-top-left-radius: 4px; border-top-right-radius: 4px; border-bottom-right-radius: 4px; border-bottom-left-radius: 4px; opacity: 0; align-self: center; background-image: url(&quot;/assets/40e8a44f9e49d1d30b81b03f7065b738.svg&quot;); width: 14px; height: 14px;"><div name="button-object-button-structure" style="margin: 0px; padding: 0px; border: 0px; outline: 0px; font-weight: inherit; font-style: inherit; font-family: inherit; font-size: 22px; vertical-align: baseline;"></div></a></h6>

FIELD | TYPE | DESCRIPTION
-- | -- | --
type | integer | 2 for a button
style | integer | one of button styles
label? | string | text that appears on the button, max 80 characters
emoji? | partial emoji | name, id, and animated
custom_id? | string | a developer-defined identifier for the button, max 100 characters
url? | string | a url for link-style buttons
disabled? | boolean | whether the button is disabled (default false)

Button Object
Button Structure

FIELD	TYPE	DESCRIPTION
type	integer	2 for a button
style	integer	one of [button styles](https://discord.com/developers/docs/interactions/message-components#button-object-button-styles)
label?	string	text that appears on the button, max 80 characters
emoji?	partial [emoji](https://discord.com/developers/docs/resources/emoji#emoji-object)	name, id, and animated
custom_id?	string	a developer-defined identifier for the button, max 100 characters
url?	string	a url for link-style buttons
disabled?	boolean	whether the button is disabled (default false)

Reply:
Yes.